edx_rest_api_client/auth.py

```diff
@@ -48,3 +48,16 @@ def __call__(self, r):
         """ Update the request headers. """
         r.headers['Authorization'] = 'Bearer {}'.format(self.token)
         return r
+
+
+class OidcIdTokenAuth(AuthBase):
+    """ Attaches Open Id Authentication to the given Request object. """
```

Comment:
As @clintonb said, this is only tangentially related to OIDC; including that name in this context is confusing. The class merely attaches a generated JWT to the request. Please rename as appropriate.